### DISCUSSION TITLE
Solution diverges with some liquid phase reactions

### DISCUSSION THREAD

**Discussion by Jae-Nam** [shannon-lab/zapdos#103] [Q&A Plasma Chemistry / CRANE]
Hello all,

I need your help in solving Helium plasma - liquid anode interaction problem. 

To describe problem briefly, 
1. I modified rate_units.i (from "tests/crane_action") to have Helium as a background gas, and this worked very well.
2. Then, liquid phase domain was added to this file. The solution consists of Nacl+Ferricyanide. 
3. When I added some reactions like Na^+ + e_liq => Na or Ferricyanide+e_liq=Ferrocyanide, the timestep is divided too much with these statements
![image](https://user-images.githubusercontent.com/95106125/150012336-a4891ec9-21b0-49cd-af44-f9a91a4efd5d.png)

and calculation does not proceed with below error message
![image](https://user-images.githubusercontent.com/95106125/150012360-b5b13ebd-ec7c-4a9e-9781-d7fa898ef0cb.png)

From several attempts, this problem occurs regardless of grid size. And this happens when Ferricyanide or Nacl concentration is ~mM. But I cannot find any signs of divergence in post processing tool (paraview) or residuals of variables in terminal prompt. Are there any methods of debugging this kinds of problems? or Can anybody look into this input file?

[Input_files.zip](https://github.com/shannon-lab/zapdos/files/7892035/Input_files.zip)

Thank you!

**Comment by lindsayad** [comment]
Hi, could you paste your `Executioner` block? And also the entire nonlinear solve history? (E.g. show the iterations from the beginning of the timestep up to the failure). A nonlinear residual of 1e-10 is already quite low and it's possible your solve tolerances are too tight

**Reply by Jae-Nam** [reply]
Here is Executioner block

[Executioner]
  type = Transient
  #end_time = 0.00737463126
  end_time = 1e-5
  petsc_options = '-snes_converged_reason -snes_linesearch_monitor'
  solve_type = NEWTON #NEWTON
  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount -ksp_type -snes_linesearch_minlambda'
  petsc_options_value = 'lu NONZERO 1.e-10 fgmres 1e-3'
  nl_rel_tol = 1e-08
  dtmin = 1e-16
  l_max_its = 20
  [./TimeStepper]
    type = IterationAdaptiveDT
    cutback_factor = 0.4
    dt = 1e-12
    growth_factor = 1.2
   optimal_iterations = 15
  [../]
[]

**Reply by Jae-Nam** [reply]
This is a solution history
[solution history.txt](https://github.com/shannon-lab/zapdos/files/7892175/solution.history.txt)

**Reply by lindsayad** [reply]
Try changing `nl_rel_tol` to 1e-7 and also add the option `line_search = 'none'`
